Sources/NIO/NonBlockingFileIO.swift

```diff
@@ -253,20 +253,45 @@ public struct NonBlockingFileIO {
         }
     }
 
-    /// Open the file at `path` on a private thread pool which is separate from any `EventLoop` thread.
+    /// Open the file at `path` for reading on a private thread pool which is separate from any `EventLoop` thread.
     ///
     /// This function will return (a future) of the `NIOFileHandle` associated with the file opened and a `FileRegion`
     /// comprising of the whole file. The caller must close the returned `NIOFileHandle` when it's no longer needed.
     ///
     /// - note: The reason this returns the `NIOFileHandle` and the `FileRegion` is that both the opening of a file as well as the querying of its size are blocking.
     ///
     /// - parameters:
-    ///     - path: The path of the file to be opened.
+    ///     - forReadingAtPath: The path of the file to be opened for reading.
     ///     - eventLoop: The `EventLoop` on which the returned `EventLoopFuture` will fire.
     /// - returns: An `EventLoopFuture` containing the `NIOFileHandle` and the `FileRegion` comprising the whole file.
-    public func openFile(path: String, eventLoop: EventLoop) -> EventLoopFuture<(NIOFileHandle, FileRegion)> {
+    public func openFile(forReadingAtPath path: String, eventLoop: EventLoop) -> EventLoopFuture<(NIOFileHandle, FileRegion)> {
```

Comment:
Again, this can only be changed if we either make this a semver major or return the old version and deprecate it.

Reply:
Reverted reading argument name